cli/repair/patch.go

```diff
@@ -0,0 +1,174 @@
+		}
+	}
+
+	return resMessages, nil
+}
+
+func rewriteConf(topologyConfPath string, topologyConf *TopologyConfType) ([]common.ResultMessage, error) {
+	var resMessages []common.ResultMessage
+
+	resMessages = append(resMessages, common.GetDebugMessage("Topology config file: %s", topologyConfPath))
+
+	backupPath, err := createFileBackup(topologyConfPath)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to create topology config backup: %s", err)
+	}
+	resMessages = append(resMessages, common.GetDebugMessage("Created backup file: %s", backupPath))
+
+	newConfContent, err := topologyConf.MarshalContent()
+	if err != nil {
+		return nil, fmt.Errorf("Failed to get new config content: %s", err)
+	}
+
+	confFile, err := os.OpenFile(topologyConfPath, os.O_WRONLY|os.O_TRUNC, 0755)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to open a new config: %s", err)
+	}
+
+	if _, err := confFile.Write(newConfContent); err != nil {
+		return nil, fmt.Errorf("Failed to write a new config: %s", err)
+	}
+
+	return resMessages, nil
+}
+
+func reloadConf(topologyConfPath string, instanceName string, ctx *context.Ctx) ([]common.ResultMessage, error) {
+	var resMessages []common.ResultMessage
+
+	consoleSock := project.GetInstanceConsoleSock(ctx, instanceName)
+	resMessages = append(resMessages, common.GetDebugMessage("Instance console socket: %s", consoleSock))
+
+	if _, err := os.Stat(consoleSock); err != nil {
+		return nil, fmt.Errorf("Failed to use instanace console socket: %s", err)
+	}
+
+	conn, err := net.Dial("unix", consoleSock)
+	if err != nil {
+		return resMessages, fmt.Errorf("Failed to dial: %s", err)
+	}
+
+	defer conn.Close()
+
+	// read greeting
+	if _, err := common.ReadFromConn(conn); err != nil {
+		return resMessages, fmt.Errorf("Failed to read greeting: %s", err)
+	}
+
+	// eval
+	evalFuncTmpl := `
+		local ClusterwideConfig = require('cartridge.clusterwide-config')
+		local cfg, err = ClusterwideConfig.load('{{ .ConfigPath }}')
+		if err ~= nil then
+			return nil, string.format('Failed to load new config: %s', err)
+		end
+
+		local confapplier = require('cartridge.confapplier')
+
+		local roles_configured_state = 'RolesConfigured'
+		local connecting_fullmesh_state = 'ConnectingFullmesh'
+
+		local state = confapplier.wish_state(roles_configured_state, {{ .WishStateTimeout }})
+
+		if state == connecting_fullmesh_state then
+			return nil, string.format(
+				'Failed to desire %s config state. Stuck in %s. ' ..
+					'Call "box.cfg({replication_connect_quorum = 0})" in instance console and try again',
+				roles_configured_state, state
+			)
+		end
+
+		if state ~= roles_configured_state then
+			return nil, string.format(
+				'Failed to desire %s config state. Stuck in %s',
+				roles_configured_state, state
+			)
+		end
+
+		local current_uuid = box.info().uuid
+		if cfg:get_readonly().topology.servers[current_uuid] == nil then
+			return false
+		end
+
+		cfg:lock()
```

Comment:
AFAIK it should be done internally by cartridge. You don't really need it. Also you lock config here and don't unlock it.

Reply:
I've briefly looked at cartridge code. It seems to be ok.